### ISSUE TITLE
When I use image with large size (800, 1200) or (1200, 1600), the RPN loss is large, 5.123 or nan?

### ISSUE THREAD

**Issue by hdjsjyl** [bowenc0221/Decoupled-Classification-Refinement#7]
Thanks for your excellent work. I am a little confused about image size, When I use image size (600, 1000), the loss is less than 1, but When I use image size (800, 1200), the loss is greater than 5, image size (1200, 1600) with loss greater than 10 or nan. Any advice will be appreciated. Thanks

**Comment by bowenc0221** [comment]
@hdjsjyl Can you share your config file and log file?

**Comment by hdjsjyl** [comment]
@bowenc0221 yes, the following is log information: thanks
sftp://sabine.cacds.uh.edu/brazos/kakadiaris/Experiments/lei/DecoupledClassificationRefinement2/faster_rcnn_dcr/output/wider_dcr/resnet_v1_101_wider_train2017_dcn_dcr_end2end/train/resnet_v1_101_wider_train2017_dcn_dcr_end2end_2019-01-05-16-47.txt

**Comment by hdjsjyl** [comment]
@bowenc0221 yes, the following is log file, thanks

[resnet_v1_101_wider_train2017_dcn_dcr_end2end_2019-01-05-16-47.txt](https://github.com/bowenc0221/Decoupled-Classification-Refinement/files/2730910/resnet_v1_101_wider_train2017_dcn_dcr_end2end_2019-01-05-16-47.txt)

**Comment by bowenc0221** [comment]
@hdjsjyl I see your problem. Do you know the size of objects in your new dataset? You only scale the image, but anchors (largest is 8*32=256) might be too small to match a gt box (gt box is scaled but anchor is not scaled). This is why the bbox regression loss for RPN becomes so big. Try printing the IoU between anchors and gt boxes in:
https://github.com/bowenc0221/Decoupled-Classification-Refinement/blob/9c37d014212f2e9f335a7262e3b1c24cd6774e49/lib/rpn/rpn.py#L161

**Comment by hdjsjyl** [comment]
@bowenc0221 you're correct. Before I scale the anchors, most of the IoU values are 0, other values are less than 1; after I scale the anchors, the RPNL1Loss is less than 1 and normal, but most of the IoU values are still 0, other values are less than 1. Most objects of my datasets are small objects, do you have some ideas for it? Thanks and Happy New Year!

**Comment by bowenc0221** [comment]
@hdjsjyl maybe you can calculate the distribution of object sizes in the training set and design anchor size accordingly.

**Comment by hdjsjyl** [comment]
@bowenc0221 I am doing it, thanks.
